arrows/klv/tests/test_klv_blob.cxx

```diff
@@ -0,0 +1,83 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief Test KLV blob functions.
+
+#include <arrows/klv/klv_blob.txx>
+
+#include <gtest/gtest.h>
+
+#include <vector>
+
+// ----------------------------------------------------------------------------
+int
+main( int argc, char** argv )
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  return RUN_ALL_TESTS();
+}
+
+// ----------------------------------------------------------------------------
+#define CALL_TEST( func, ... ) \
+  do { SCOPED_TRACE( #func ); func( __VA_ARGS__ ); } while( 0 )
+
+using namespace kwiver::arrows::klv;
+
+// ----------------------------------------------------------------------------
+void
+test_blob_read( klv_bytes_t const& data )
+{
+  auto it = data.cbegin();
+  EXPECT_EQ( data, *klv_read_blob( it, data.size() ) );
+  EXPECT_EQ( data.cend(), it );
+}
+
+// ----------------------------------------------------------------------------
+TEST( klv, blob_read )
+{
+  CALL_TEST( test_blob_read, {} );
+  CALL_TEST( test_blob_read, { 0x00 } );
+  CALL_TEST( test_blob_read, { 0xFF, 0xFF } );
+  CALL_TEST( test_blob_read,
+             { 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88, 0x99, 0x00 } );
+}
+
+// ----------------------------------------------------------------------------
+void
+test_blob_write( klv_bytes_t const& data )
+{
+  klv_bytes_t buffer( data.size(), 0xba );
```

Comment:
```suggestion
  klv_bytes_t buffer{ data.size(), 0xba };
```

Reply:
This one gives me an error; I think with the braces it tries to interpret the arguments as an initializer list of integers, not as separate arguments to a constructor.

Reply:
Ah, right, `std::vector` constructors, how we hate thee...